sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_init_test.c

```diff
@@ -42,15 +41,18 @@ rom_error_t retention_ram_init_test(void) {
   } else if (bitfield_bit32_read(reset_reasons, kRstmgrReasonSoftwareRequest)) {
     // This branch runs after the SW-requested reset
     LOG_INFO("Ensuring all sections have changed");
-    memcpy(raw, ret, sizeof(retention_sram_t));
 
+    static_assert(sizeof(retention_sram_t) % sizeof(uint64_t) == 0,
+                  "This test expects the retention SRAM size to be a multiple "
+                  "of uint64_t");
     uint32_t matches = 0;
-    for (size_t i = 0; i < ARRAYSIZE(raw); ++i) {
-      if (raw[i] == pattern64) {
+    for (size_t i = 0; i < sizeof(retention_sram_t) / sizeof(uint64_t); i++) {
+      if (read_64((char *)ret + i * sizeof(uint64_t)) == pattern64) {
```

Comment:
nit, just to make things simpler:
```suggestion
    for (size_t i = 0; i < sizeof(retention_sram_t); i += sizeof(uint64_t)) {
      if (read_64((char *)ret + i) == pattern64) {
```

Reply:
Done